### ISSUE TITLE
Missing leading 0's when using to_json_array

### ISSUE THREAD

**Issue by matthewdear** [beached/daw_json_link#439]
input json: [
	{
		"i":5,
		"d":0.002,
		"b":false,
		"s":"hello world",
		"y":[1,2,3,4] 
    },
	{
		"i":4,
		"d":1.000005,
		"b":true,
		"s":"goodbye world",
		"y":[4,3,1,4] 
    }]

It parses the JSON correctly

but when I parse the object back to json using to_json_array, the result is:
[
{"i":5,"d":0.002,"b":false,"s":"hello world","y":[1,2,3,4]},
{"i":4,"d":1.5,"b":true,"s":"goodbye world","y":[4,3,1,4]}
]

element d in the second array, 1.000005 gets parsed back as 1.5, which is incorrect.
however the in the first array 0.002 gets parsed back correctly as 0.002.

Is there something wrong with the below code or is this a bug in the library?

My Code:
```

struct TestClass {
	int i;
	double d ;
	bool b ;
	std::string s{};
	std::vector<int> y{};

	TestClass(int Int, double Double, bool Bool, std::string S,
		std::vector<int> Y)
		: i(Int), d(Double), b(Bool), s(std::move(S)), y(std::move(Y)) {}
};

namespace daw::json {
	template <>
	struct json_data_contract<TestClass> {
		using type =
			json_member_list<
			json_number<"i", int>,
			json_number<"d">,
			json_bool<"b">,
			json_string<"s">,
			json_array<"y", int>
			>;

		static inline auto to_json_data(TestClass const & value)
		{
			return std::forward_as_tuple(
				value.i,
				value.d,
				value.b,
				value.s,
				value.y
				);

		};
	};
} 


TEST(TestJsonParse, TestTestParse)
{
	std::string_view json_array_data = R"([
	{
		"i":5,
		"d":0.002,
		"b":false,
		"s":"hello world",
		"y":[1,2,3,4] 
    },
	{
		"i":4,
		"d":1.000005,
		"b":true,
		"s":"goodbye world",
		"y":[4,3,1,4] 
    }])";

	std::vector<TestClass> arry_of_test_class =
		daw::json::from_json_array<TestClass>(json_array_data);

	auto json_array = daw::json::to_json_array(arry_of_test_class);

}

```

**Comment by beached** [comment]
The issue looks like it is in the to_chars implementation code

**Comment by beached** [comment]
When formatting the output, it fails here 
https://github.com/beached/daw_json_link/blob/release/include/daw/json/impl/to_daw_json_string.h#L1536
the sig digits of `1.000005` is `1000005` and it subtracts `1000000` from it to get access to the fractional parts

**Comment by beached** [comment]
Fixed by #440

**Comment by beached** [comment]
Thanks for finding this. :)
